docs/source/custom/package.rst

```diff
@@ -1,17 +1,82 @@
-==========================
-Packaging and Distribution
-==========================
+=================================
+Using and Distributing Extensions
+=================================
 
-Ideally, any custom code is made available as a regular Python package.
+Extensions for :py:mod:`cobald` are regular Python code accessible to the interpreter.
+For specific problems, extensions can be defined directly in a Python configuration file.
+General purpose and reusable code should be made available as a regular Python package.
 This ensures proper installation and dependency management.
-Once installed, :py:mod:`cobald` can access and use such packages directly.
 
-Writing the setup.py
---------------------
+Configuration Files
+===================
 
-Packages for :py:mod:`cobald` can follow the `PyPA`_ recommendations for `python packaging`_.
-Their ``setup.py`` file should contain a dependency on :py:mod:`cobald`.
-Ideally, the keywords should contain `cobald` for findability.
+Using Python :doc:`configuration files </source/daemon/config>` allows to define arbitrary
+objects, functions and helpers.
+This is ideal for minor modifications of existing objects and
+experimental extensions.
+Simply add new definitions to the configuration before using them:
+
+.. code:: python3
+
+    #/etc/cobald/my_demo.py
+    from cobald.interface import Controller
+
+    from cobald_demo.cpu_pool import CpuPool
+    from cobald_demo.draw_line import DrawLineHook
+
+
+    # custom Controller implementation
+    class StaticController(Controller):
+        """Controller that sets demand to a fixed value"""
+        def __init__(self, target, demand):
+            super().__init__(target)
+            self.target.demand = demand
+
+    # use custom Controller
+    pipeline = StaticController.s(demand=50) >> DrawLineHook.s() >> CpuPool(interval=1)
+
+Configuration files are easy to use and modify, but impractical for reusable extensions.
+
+Python Packages
+===============
+
+For generic extensions, Python packages simplify distribution and reuse.
+Packages are individual `.py` files or folders containing several `.py` files;
+in addition, packages contain meta-data for dependency management and installation.
+
+.. code:: python3
+
+    # my_controller.py
+    from cobald.interfaces import Controller
+
+    class StaticController(Controller):
+        def __init__(self, target, demand):
+            super().__init__(target)
+            self.target.demand = demand
+
+Packages can be temporarily mounted via ``PYTHONPATH`` or permanently installed.
```

Comment:
Is it correct here to talk about *mounting*? I feel like *loading* might be better?

```suggestion
Packages can be temporarily loaded via ``PYTHONPATH`` or permanently installed.
```

Reply:
```suggestion
Packages can be temporarily accessed via ``PYTHONPATH`` or permanently installed.
```